File: backend/app/models/checklist.py
import uuid

from sqlalchemy import Boolean, DateTime, ForeignKey, Integer, String, func
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base

CHECKLIST_CATEGORIES = ["Packing", "Documents", "Preparation", "Other"]


class ChecklistItem(Base):
    __tablename__ = "checklist_items"

    id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    trip_id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), ForeignKey("trips.id", ondelete="CASCADE"), nullable=False)
    title: Mapped[str] = mapped_column(String, nullable=False)
    is_checked: Mapped[bool] = mapped_column(Boolean, nullable=False, default=False)
    category: Mapped[str] = mapped_column(String, nullable=False, default="Other")
    order_index: Mapped[int] = mapped_column(Integer, nullable=False, default=0)
    created_at: Mapped[DateTime] = mapped_column(DateTime(timezone=True), server_default=func.now())

File: frontend/types/checklist.ts
export const CHECKLIST_CATEGORIES = ["Packing", "Documents", "Preparation", "Other"] as const;
export type ChecklistCategory = typeof CHECKLIST_CATEGORIES[number];

export interface ChecklistItem {
  id: string;
  trip_id: string;
  title: string;
  is_checked: boolean;
  category: string;
  order_index: number;
  created_at: string;
}

export interface ChecklistItemCreate {
  title: string;
  category?: string;
  is_checked?: boolean;
}

export interface ChecklistItemUpdate {
  title?: string;
  category?: string;
  is_checked?: boolean;
}
